python/dgl/nn/pytorch/explain/gnnexplainer.py

```diff
@@ -425,3 +398,395 @@ def explain_graph(self, graph, feat, **kwargs):
+    It identifies compact subgraph structures and small subsets of node features that play a
+    critical role in GNN-based node classification and graph classification.
+    To generate an explanation, it learns an edge mask :math:`M` and a feature mask :math:`F`
+    by optimizing the following objective function.
+    .. math::
+      l(y, \hat{y}) + \alpha_1 \|M\|_1 + \alpha_2 H(M) + \beta_1 \|F\|_1 + \beta_2 H(F)
+    where :math:`l` is the loss function, :math:`y` is the original model prediction,
+    :math:`\hat{y}` is the model prediction with the edge and feature mask applied, :math:`H` is
+    the entropy function.
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain.
+        * The required arguments of its forward function are graph and feat.
+          The latter one is for input node features.
+        * It should also optionally take an eweight argument for edge weights
+          and multiply the messages by it in message passing.
+        * The output of its forward function is the logits for the predicted
+          node/graph classes.
+        See also the example in :func:`explain_node` and :func:`explain_graph`.
+    num_hops : int
+        The number of hops for GNN information aggregation.
+    lr : float, optional
+        The learning rate to use, default to 0.01.
+    num_epochs : int, optional
+        The number of epochs to train.
+    alpha1 : float, optional
+        A higher value will make the explanation edge masks more sparse by decreasing
+        the sum of the edge mask.
+    alpha2 : float, optional
+        A higher value will make the explanation edge masks more sparse by decreasing
+        the entropy of the edge mask.
+    beta1 : float, optional
+        A higher value will make the explanation node feature masks more sparse by
+        decreasing the mean of the node feature mask.
+    beta2 : float, optional
+        A higher value will make the explanation node feature masks more sparse by
+        decreasing the entropy of the node feature mask.
+    log : bool, optional
+        If True, it will log the computation process, default to True.
+    """
+    
+    def __init__(self, 
+                 model, 
+                 num_hops, 
+                 lr=0.01, 
+                 num_epochs=100, 
+                 *,
+                 alpha1=0.005,
+                 alpha2=1.0,
+                 beta1=1.0,
+                 beta2=0.1,
+                 log=True):
+        super(HeteroGNNExplainer, self).__init__()
+        self.model = model
+        self.num_hops = num_hops
+        self.lr = lr
+        self.num_epochs = num_epochs
+        self.alpha1 = alpha1
+        self.alpha2 = alpha2
+        self.beta1 = beta1
+        self.beta2 = beta2
+        self.log = log
+
+    def _init_masks(self, graph, feat):
+        r"""Initialize learnable feature and edge mask.
+        Parameters
+        ----------
+        graph : DGLHeteroGraph
```

Comment:
`DGLGraph` is an alias for `DGLHeteroGraph`. Let's keep using it for consistency.